File: LeetCode/Medium/2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
from collections import deque

class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        n = len(part)
        q = deque([])

        for c in s:
            q.append(c)
            if len(q) >= n:
                comp_str = "".join(list(q)[-n:])
                if comp_str == part:
                    for _ in range(n):
                        q.pop()

        return "".join(list(q))

File: LeetCode/Medium/2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.rs
impl Solution {
    pub fn remove_occurrences(s: String, part: String) -> String {
        let mut chars: Vec<char> = Vec::new();

        for c in s.chars() {
            chars.push(c);

            if chars.len() >= part.len() {
                let comp_str: String = chars[chars.len() - part.len()..].iter().collect();

                if comp_str == part {
                    for _ in 0..part.len() {
                        chars.pop();
                    }
                }
            }
        }

        chars.iter().collect()
    }

}